src/hooks/useHashConnect/useHashConnect.ts

```diff
@@ -59,7 +59,13 @@ const useHashConnect = ({
   useHashConnectEvents(hashconnect, dispatch, debug);
 
   const saveToLocalStorage = useCallback(() => {
-    const hashconnectDataJSON = JSON.stringify(hashConnectState);
+    const hashconnectDataJSON = JSON.stringify({
+      ...hashConnectState,
+      // do not save non wallet related state fields
+      transactionState: { ...initialHashConnectState.transactionState },
```

Comment:
I don't think there is an issue with this logic but completely removing these extra fields from local storage instead of setting them to their initial value would be a safer operation. It also makes it clear to other devs that those fields should not be read from local storage. You could use the Ramada **omit** function to remove the excess fields.

Reply:
resolving in this next commit. doing this though i had to make another tweak in initHashConnectReducer because it was missing the non wallet specific fields we removed from local storage, causing undefined errors when referencing these fields. to resolve i basically joined the fields (that we omit from local storage) in initialhashConnectState with the object in local storage. I'll leave a comment after i push in the line where its at